dts/arm/raspberrypi/rpi_pico/rp2350.dtsi

```diff
@@ -456,6 +456,23 @@
 			interrupt-names = "irq0", "irq1";
 			status = "disabled";
 		};
+
+		ipm0: sio-fifo@d0000000 {
```

Comment:
Other drivers cannot use the address range written here.
In other words, if you create a driver that uses other SIO addresses,
an address overlap will occur, and a DTS compilation error will occur.
It is best to limit the address range to that used by the FIFO.
(In fact, pico-sdk functions can escape this restriction. But expecting it is bad design.)

Reply:
Yes. Otherwise, it would be impossible to coexist with drivers for other functions.
In fact, the pico-sdk implementation doesn't take DTS into account, so it's just a formal check.